app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/BankAccountController.scala

```diff
@@ -240,11 +240,10 @@ object BankAccountController {
       .transform[AccountNumber](s => AccountNumber(s.replaceAllLiterally(" ", "")), _.value)
       .verifying("invalid", e => accountNumberRegex.test(e.value))
 
-  val sortCodeRegex: Predicate[String]   = "^\\d{6}$".r.pattern.asPredicate()
   val sortCodeMapping: Mapping[SortCode] =
-    nonEmptyText(minLength = 6, maxLength = 6)
-      .transform[SortCode](s => SortCode(s.replaceAllLiterally(" ", "")), _.value)
-      .verifying("invalid", e => sortCodeRegex.test(e.value))
+    nonEmptyText
+      .verifying("invalid", str => SortCode.isValid(str))
```

Comment:
How about you transform first and then verify?
By doing so, you will not need to re-implement your regex twice!
As an example see: PhoneNumber